kubernetes/MustacheTemplate/resource.yaml

```diff
@@ -244,9 +246,14 @@ items:
                                   type:
                                     type: string
                                     enum: [number, boolean, json, jsonString, base64]
+                                  matchLabels:
+                                    type: object
+                                    additionalProperties: true
```

Comment:
i cant seem to find documentation on additionalProperties, so im not 100% sure about what im about to say.

In the past, in order to avoid pruning unknown fields, i would specify `x-kubernetes-preserve-unknown-fields: true`. Do you still need to do that, or does `additionalProperties: true` handle that for you?

Reply:
Hmm. I had not seen that before [additionalProperties](http://json-schema.org/understanding-json-schema/reference/object.html#additional-properties) is just part of the open api / json schema spec.

It lets me do this and it preserves the properties I set. I'm using kubernetes 1.20. maybe it is a newer feature

Reply:
added it as well. I'm not sure that its doing anything, but is also valid